apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -179,10 +189,10 @@ where
     #[serde(skip)]
     defer_on_tail: Option<DeferDirectiveArguments>,
     /// At the point where a `@context` is set, we will have fields to select
-    context_to_selection: Vec<Option<IndexSet<String>>>,
+    context_to_selection: Vec<Option<ContextToSelection>>,
```

Comment:
Does this being `None` mean something else than it being an empty set inside `Some()`? If not, we can do without the option which will make it easier to use everywhere

Reply:
See my comment to Tyler below. Yes, None does have special meaning because all of the vectors need to be the same length. Each index translates to a `GraphPathItem`, but a `GraphPathItem` may or may not have values set for each of these things.

```rust
pub(crate) type GraphPathItem<'path, TTrigger, TEdge> = (
    TEdge,
    &'path Arc<TTrigger>,
    &'path Option<Arc<OpPathTree>>,
    Option<ContextToSelection>,
    Option<ParameterToContext>,
);```